docs/how_it_works.rst

```diff
@@ -14,38 +39,57 @@ these as *passive* and *active* discovery.
+      Node.js "bootstrap" script from RackHD. RackHD runs the bootstrap program which uses a simple REST
+      API to "ask" what it should do on the remote host.
 
-1. Discovery is initiated by sending down the iPXE boot loader with a pre-built script to run within iPXE. This script then chainloads into a new, dynamically rendered iPXE script that interrogates the enabled network interfaces on the remote machine and reports them back to RackHD. RackHD adds this information to the machine and lookup records.
+   b) The workflow engine, running the discovery
+      workflow, provides a set of tasks to run. These tasks are matched with parsers in RackHD to understand
+      and store the output. They work together to run Linux commands that interrogate the hardware from the
+      microkernel running in memory. These commands include interrogating the machine's BMC settings through
+      IPMI, the installed PCI cards, the DMI information embedded in the BIOS, and others. The resulting
+      information is then stored in JSON format as "catalogs" in RackHD.
 
-2. RackHD then renders an additional iPXE script to be chainloaded that downloads and runs the microkernel. The microkernel boots up and requests a Node.js "bootstrap" script from RackHD. RackHD runs the bootstrap program which uses a simple REST API to "ask" what it should do on the remote host. The workflow engine, running the discovery workflow, provides a set of tasks to run. These tasks are matched with parsers in RackHD to understand and store the output. They work together to run Linux commands that interrogate the hardware from the microkernel running in memory. These commands include interrogating the machine's BMC settings through IPMI, the installed PCI cards, the DMI information embedded in the BIOS, and others. The resulting information is then stored in JSON format as "catalogs" in RackHD.
+   c) When it's completed with all the tasks, it tells the microkernel to reboot the machine and sends an
+      internal event that the basic bootstrapping process is finished
 
-3. The discovery workflow then performs a workflow task process called "SKU analysis" that compares the catalog data for the node against SKU definitions
-   loaded into the system through the REST interface. If the definitions match, RackHD updates its data model indicating that the node belongs to a SKU.
+2. The SKU Discovery workflow then performs a workflow task process called "generate-sku" that compares the
+   catalog data for the node against SKU definition loaded into the system through the REST interface. If
+   the definitions match, RackHD updates its data model indicating that the node belongs to a SKU.
 
-4. The workflow uses the IPMI channels to set up a connection to the BMC so that it can control that node in the future (power on, off and reboot).
+4. The task "generate-enclosure" uses the IPMI channels to set up a connection to the BMC so that it can control that
```

Comment:
It was for a while the conductor with hardcoded stuff. We have a workflow called `create-ipmi-obm-settings` that can be run based on SKU definitions, and takes a username/password to update the node document, but nothing is done by default at present.